proposals/LP0004-project-governance.md

```diff
@@ -0,0 +1,393 @@
+
+Both the _community_ and _infrastructure area teams_ will select one member of their team to act as a liaison with the LLVM Foundation. The liaisons will be responsible for managing the relationship between their _area team_ and the employees and directors of the LLVM Foundation.
+
+##### Voting by Area
+
+Every _voting contributor_ is eligible to vote for the project-wide _Community_ and _Infrastructure_ teams. Eligibility for voting on other _area teams_ is determined by _meaningful contributions_ to the areas those teams cover. For example, in order to vote on the **LLVM Backend** _area team_, a contributor would need to contribute code to the LLVM `CodeGen`, `MC`, or `Target` libraries, or to vote on the members of the **Flang** team, a contributor would need to contribute code to the `flang` project. There is no limit to the number of _area teams_ a contributor can vote on if they have made contributions to the relevant areas.
+
+#### Vacancies
+
+A member of an _area team_ can resign at any time. Additionally as life can sometimes happen unexpectedly a member of an _area team_ may be unable to fulfill their duties or resign. In that case a majority of the remaining _area team_ may vote to declare the member removed in absentia after a 90-day absence.
+
+If someone resigns or is otherwise removed from an _area team_, the remaining members of the _area team_ may appoint a replacement to serve the remainder of the term through any process they choose.
+
+A removed member of an _area team’s_ full term is counted toward term limits, not just the time served. For an appointed member the time acting counts toward term limits, but should not impair the ability to run and serve two complete concurrent terms, unless they were previously term-limited in which case, being appointed to serve a partial term does not allow for the 1 year break between terms.
+
+### Project Council
+
+The last structure this proposal defines is the _project council_. The _project council_ is composed of the chair from each of the _area teams_.
+
+The _project council_ has a mandate to:
+
+* Prioritize the long term health of the LLVM project and community.
+* Shape the community to be accessible, inclusive, and sustainable.
+* Maintain the relationship between the LLVM Community and the LLVM Foundation.
+* Facilitate seeking consensus among the LLVM Community and _area teams_.
+* As a last resort, act as the final decision maker on debates.
+
+The _project council_ has the power to form and dissolve _area teams_. Forming an _area team_ requires a majority vote. Dissolving an _area team_, or altering the boundaries of an _area team_ requires a consenting vote of the chair of the _area team(s)_ being altered and a majority vote of the _project council_.
```

Comment:
If I read this correctly, the representative of the area team has a veto against modifying or dissolving this. I can see how that makes sense if the area team wants to voluntarily modify or dissolve itself. What kind of power does the Project Council have in the involuntary case? Is the idea that if an area team is not holding itself to its mandate, everything is resolved through the CoC? I asked earlier about who's keeping the area teams accountable, and with the powers outlined here, it doesn't appear to be the project council.

Reply:
What about adding something like:

> If the _project council_ is seeking to dissolve an _area team_ and the chair of that team does not consent, the team may be dissolved with a unanimous vote of the remaining _project council_ members only after consultation with the Code of Conduct Committee to ensure that all project policies are appropriately followed.